configuration/yaml.md

```diff
@@ -0,0 +1,338 @@
+    groups:
+      - lBedroom1
+    tags:
+      - Lightbulb
+
+  BedRoom1_Light_Switch:
+    type: Switch
+    label: Bed Room1 Light Switch
+    icon: switch
+    format: '%s'
+    groups:
+      - BedRoom1_Light_Bulb
+    tags:
+      - Switch
+    channel: mqtt:topic:bedroom1-switch:switch1
+
+  BedRoom1_Switch_StatusLED:
+    type: Switch
+    groups:
+      - BedRoom1_Switch_Group
+    channels:
+      mqtt:topic:bedroom1-switch:statusled:
+        # The configuration of this item-channel link is defined below
+        profile: transform:MAP
+        function: "|open=ON;closed=OFF" # | is a YAML special character, so enclose it in quotes.
+```
+
+> Note:
+>
+> The last item-channel link in the example above requires some configurations,
+> so it needs to be defined with `channels:` instead of the short form `channel:`.
+> In this instance, it is configured to use a Profile,
+> specifically a MAP Profile.
+> The item-channel configuration keys and values depend on which profile is used, and
+> the details can be found in the corresponding profile's documentation.
+> For example, [MAP Profile]({{base}}/addons/transformations/map/#usage-as-a-profile) requires
+> a `function` parameter.
+>
+> For more information on Profiles, see:
+>
+> - [Item-Channel Link Profile]({{base}}/configuration/items.html#profiles)
+> - [Script Transformation Profile]({{base}}/configuration/transformations.html#script-transformation-profile)
+> - List of available [transfomation addons](https://www.openhab.org/addons/#transform), most of which support profiles
+
+### Tags
+
+openHAB comes with pre-defined [semantic tags](https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.semantics/model/SemanticTags.csv) that can be readily used.
+It is also possible to add custom semantic tags into the system and subsequently use them alongside the built-in tags.
+
+Custom semantic tags can be defined under the `tags` section.
+
+#### Tags YAML Definition Syntax
+
+```yaml
+tags:
+  <tag_full_uid>:
+    label: <tag label>
+    description: <tag description>
+    synonyms:
+      - tag synonym 1
+      - tag synonym 2
+```
+
+Each tag is specified with the tag's full UID as the key.
+The rest are optional.
+
+Example:
+
+```yaml
+version: 1
+
+tags:
+  Location_Indoor_Room_HomeCinemaRoom:
+    label: Home Cinema Room
+    description: The room containing my home cinema system
+    synonyms:
+      - Home theater room
+
+  Location_Outdoor_VegetableGarden:
+    label: Vegetable Garden
+
+  Property_Height: {} # no label/description/synonyms
+```
+
+The tag's full UID defines the location of the semantic tag in the tree.
+In the example, `Location_Indoor_Room_HomeCinemaRoom` defines semantic tag `HomeCinemaRoom` as a child of `Room` which is a child of `Indoor` which is a child of `Location`.
+There are constraints on the name of a semantic tag (like HomeCinemaRoom): it must start with a capital letter and contain only letters and numbers.
```

Comment:
Note that I don't see a check in the tag provider. Something to be enhanced.

Reply:
I assume this requires no changes in this PR, so marking this as resolved.